qp/ensemble.py

```diff
@@ -0,0 +1,519 @@
+        vb: boolean, optional
+            report on progress
+        where: string
+            path to file corresponding to Ensemble, presumed to not yet exist
+        procs: int, optional
+            limit the number of processors used, otherwise use all available
+        """
+        if procs is not None:
+            self.n_procs = procs
+        else:
+            self.n_procs = psutil.cpu_count()
+        self.pool = Pool(self.n_procs)
+
+        self.n_pdfs = N
+        self.pdf_range = range(N)
+
+        if truth is None:
+            self.truth = [None] * N
+        else:
+            self.truth = truth
+        if samples is None:
+            self.samples = [None] * N
+        else:
+            self.samples = samples
+        if quantiles is None:
+            self.quantiles = [None] * N
+        else:
+            self.quantiles = [(quantiles[0], quantiles[1][i]) for i in self.pdf_range]
+        if histogram is None:
+            self.histogram = [None] * N
+        else:
+            self.histogram = [(histogram[0], histogram[1][i]) for i in self.pdf_range]
+        if gridded is None:
+            self.gridded = [None] * N
+        else:
+            self.gridded = [(gridded[0], gridded[1][i]) for i in self.pdf_range]
+        self.mix_mod = None
+        self.evaluated = None
+
+        self.scheme = scheme
+
+        if vb and self.truth is None and self.quantiles is None and self.histogram is None and self.gridded is None and self.samples is None:
+            print 'Warning: initializing an Ensemble object without inputs'
+            return
+
+        self.make_pdfs()
+
+        # self.logfilename = 'logfile.txt'
+
+        self.stacked = {}
+
+        # self.where = where
+        # if os.path.isfile(self.where):
+        #     #something like this, uses connection?
+        #     self.engine = self.read(self.where)
+        # else:
+        #     self.engine = sqlalchemy.create_engine('sql:///'+self.where)
+        #
+        # self.metadata = MetaData(bind=self.engine)
+        #
+        # parametrizations_table =Table('parametrizations', metadata,)
+
+    #
+    #
+    # def read(self):
+    #
+    # def write(self):
+    #
+    # def make_db(self, where):
+    #     """
+    #     Makes a fresh set of ensemble tables
+    #     """
+    #     return
+    #
+    # def __add__(self, ensemble):
+    #
+    # def add_PDFs(self, PDFs):
+    #     """
+    #     Adds qp.PDF objects to the ensemble
+    #
+    #     Parameters
+    #     ----------
+    #     PDFs: list, qp.PDF object
+    #         list of PDF objects to add to the ensemble
+    #     """
+    #     pdf_range = range(len(PDFs))
+    #     def add_one(i):
+    #
+    #
+    #     self.pdfs = self.pool.map(add_one, pdf_range)
+    #
+    #     return(self)
+    #
+    # def add_parameterization(self, type, parameters):
+    #
+
+    def make_pdfs(self):
+        """
+        Makes a list of qp.PDF objects based on input
+        """
```

Comment:
Maybe add a Note that this function is structured so as to make good use of your multiprocessing pool?